### PR TITLE
Improve TinyYoloV3 postprocess to properly set the bounding boxes

### PR DESCRIPTION
**Pull request by GFallasRR**

### CODE REVIEW

File: ext/r2inference/gsttinyyolov3.c [hunk 1]
```diff
@@ -267,52 +266,43 @@ gst_tinyyolov3_get_boxes_from_prediction (GstTinyyolov3 * tinyyolov3,
...
+          max_class_prob = cur_class_prob;
+          max_class_prob_index = c;
         }
       }
+
+      if (max_class_prob > tinyyolov3->prob_thresh) {
+        BBox result;
+        result.label = max_class_prob_index;
+        result.prob = max_class_prob;
+        result.x = ((gfloat *) prediction)[index];
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolov3.c]: I believe Yolo gives the `(x,y)` as the center. Shouldn't we subtract `width/2` and `height/2`?

File: ext/r2inference/gsttinyyolov3.c [hunk 2]
```diff
@@ -267,52 +266,43 @@ gst_tinyyolov3_get_boxes_from_prediction (GstTinyyolov3 * tinyyolov3,
...
+        result.y = ((gfloat *) prediction)[index + 1];
+        result.width = ((gfloat *) prediction)[index + 2] - result.x;
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolov3.c]: Why are we subtracting this?

File: ext/r2inference/gsttinyyolov3.c (hunk 1 above)
Comment (GFallasRR) [reply ext/r2inference/gsttinyyolov3.c]: The output from the tested model gives top left and bottom right as coordinates from the bounding box. X and Y coordinates can be used directly from the output.

File: ext/r2inference/gsttinyyolov3.c (hunk 2 above)
Comment (GFallasRR) [reply ext/r2inference/gsttinyyolov3.c]: The tested TinyYoloV3 model gives top left and bottom right coordinates from the bounding box.
The width and height should be computed from the coordinates.
width = bottom_right_x - top_left_x
height = bottom_y - top_x

Coordinate (0,0) is top left corner.

File: ext/r2inference/gsttinyyolov3.c [hunk 3]
```diff
@@ -45,32 +45,33 @@
...
 
+/* Channels per pixel */
 #define MODEL_CHANNELS 3
-#define GRID_H 13
-#define GRID_W 13
-/* Grid cell size in pixels */
-#define GRID_SIZE 32
+/* Total boxes per image */
+/* Defined as 13 x 13 x 15 = 2535 */
+/* Grid width x Grid size x bounding boxes per grid */
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolov3.c]: Grid width x Grid height x bounding boxes per grid

File: ext/r2inference/gsttinyyolov3.c [hunk 4]
```diff
@@ -45,32 +45,33 @@
...
 /* Number of classes */
 #define CLASSES 80
-/* Number of boxes per cell */
-#define BOXES 5
+/* Number of dimensions per box */
+#define BOX_DIM 5
 /* 
  * Box dim 
  * [0]: x
  * [0]: y
```
Comment (michaelgruner) [inline ext/r2inference/gsttinyyolov3.c]: 1

**Pull request merged**